caikit/runtime/model_management/model_manager.py

```diff
@@ -381,6 +381,12 @@ def retrieve_model(self, model_id: str) -> ModuleBase:
             log.debug2(
                 "Lazy loading local model %s from %s", model_id, local_model_path
             )
+            # If the model is not present on disk, attempt to lazy load it
+            # anyway. This allows auto-finders that can infer the model's config
+            # to load based on the ID
+            if not os.path.exists(local_model_path):
```

Comment:
super nit: For clarity perspective, should we add in the above comment that in case the `local_model_path` doesn't exist, we will be using `model_id` as a backup to let underlying module impl make use of that as required.

Reply:
Also wondering if making using `model_id` as `local_model_path` can open up some security concern?

Reply:
good call on the comment

Reply:
>Also wondering if making using model_id as local_model_path can open up some security concern?

Hmm, that's interesting. I suppose it's _possible_ that an attacker could probe the filesystem layout by putting absolute paths in as `model_id` and looking at what kind of errors could come. I don't _think_ our errors would be informative enough on the user-side to be able to tell the difference between "that doesn't exist on disk" and "that's not a valid path to a model but it does exist." Even with that information, I don't think there's any way this could result in code execution without control over the `finder/initializer` code, so I don't think this is a big concern. It's also not at all new with this PR since `os.path.join("/foo", "/bar/baz")` results in just `"/bar/baz"` since it's absolute already.